modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/entries/display/context/test/ObjectEntryDisplayContextTest.java

```diff
@@ -478,44 +493,105 @@ private MockHttpServletRequest _getMockHttpServletRequest(
+					objectRelationship.getObjectDefinitionId1());
 
 			apiURL = StringBundler.concat(
-				"/o", objectDefinition.getRESTContextPath(), "/scopes/",
-				_group.getGroupId(), "/by-external-reference-code/",
+				"/o", objectDefinition.getRESTContextPath(),
+				"/by-external-reference-code/",
 				parentObjectEntry.getExternalReferenceCode(), "/",
 				objectRelationship.getName());
+
+			if (Objects.equals(
+					objectDefinition.getScope(),
+					ObjectDefinitionConstants.SCOPE_SITE)) {
+
+				apiURL = StringBundler.concat(
+					"/o", objectDefinition.getRESTContextPath(), "/scopes/",
+					_group.getGroupId(), "/by-external-reference-code/",
+					parentObjectEntry.getExternalReferenceCode(), "/",
+					objectRelationship.getName());
+			}
+
+			mockHttpServletRequest = _getMockHttpServletRequest(
+				objectEntry.getExternalReferenceCode(),
+				_objectDefinitionLocalService.getObjectDefinition(
+					objectRelationship.getObjectDefinitionId2()),
+				objectRelationship.getObjectRelationshipId(),
+				parentObjectEntry.getExternalReferenceCode());
+		}
+		else {
+			objectDefinition = _addObjectDefinition(
+				ObjectDefinitionConstants.SCOPE_DEPOT);
+
+			DepotEntry depotEntry = _depotEntryLocalService.addDepotEntry(
+				RandomTestUtil.randomLocaleStringMap(),
+				RandomTestUtil.randomLocaleStringMap(),
+				DepotConstants.TYPE_ASSET_LIBRARY,
+				ServiceContextTestUtil.getServiceContext());
+
+			_objectDefinitionSettingLocalService.addObjectDefinitionSetting(
+				objectDefinition.getUserId(),
+				objectDefinition.getObjectDefinitionId(),
+				ObjectDefinitionSettingConstants.NAME_ACCEPTED_GROUP_IDS,
+				String.valueOf(depotEntry.getGroupId()));
+
+			objectEntry = _addObjectEntry(
+				objectDefinition, String.valueOf(depotEntry.getGroupId()));
+
+			apiURL = "/o" + objectDefinition.getRESTContextPath();
+
+			if (Objects.equals(
+					objectDefinition.getScope(),
+					ObjectDefinitionConstants.SCOPE_DEPOT) ||
+				Objects.equals(
+					objectDefinition.getScope(),
+					ObjectDefinitionConstants.SCOPE_SITE)) {
+
+				apiURL += "/scopes/" + objectEntry.getScopeId();
+			}
+
+			if (objectEntry.getExternalReferenceCode() != null) {
+				apiURL +=
+					"/by-external-reference-code/" +
+						objectEntry.getExternalReferenceCode();
+			}
+
+			mockHttpServletRequest = _getMockHttpServletRequest(
+				objectEntry.getExternalReferenceCode(), LocaleUtil.getDefault(),
+				objectDefinition, 0L, null, objectEntry.getScopeId());
 		}
 
-		Assert.assertEquals(
-			apiURL,
-			_getAPIURL(
-				_getMockHttpServletRequest(
-					objectEntry.getExternalReferenceCode(),
-					_objectDefinitionLocalService.getObjectDefinition(
-						objectRelationship.getObjectDefinitionId2()),
-					objectRelationship.getObjectRelationshipId(),
-					parentObjectEntry.getExternalReferenceCode())));
+		if (!hasRelationShip) {
+			Assert.assertEquals(
+				apiURL, _getAPIURL(mockHttpServletRequest, true));
+
+			return;
+		}
```

Comment:
You added too many conditions here.
First, you check `hasRelationship`, and then for each case, you add more conditions about `scope` and `ERC`.
This creates too many layers and paths, making the logic harder to understand.

I think you could evolve this existing method to work more like what you did in the else condition: gradually building the apiURL while separating the conditions clearly:
- If there’s a relationship, concatenate this element.
- If it’s scoped by site or depot, concatenate that other element.